### ISSUE TITLE
argocd app delete is stuck with 'CleanUpAfterDeletion' enabled

### ISSUE THREAD

**Issue by mingmcb** [streamnative/pulsar-resources-operator#160]
when use argocd deploy the helm chart, we're not be able to delete the app when  'CleanUpAfterDeletion' enabled. 

the work around is to set the the argocd finalizers: to use `resources-finalizer.argocd.argoproj.io/background`.

Please take a look.

**Comment by ericsyh** [comment]
@mingmcb Can you share the manifests from the helm chart?

**Comment by ericsyh** [comment]
I guess this is caused by https://github.com/streamnative/pulsar-resources-operator/issues/69
For now, resource operator is using an all-in-one controller, and also the topic, namespace currently has no dependency management https://github.com/streamnative/pulsar-resources-operator/issues/130 which usually cause resource blocked during the deleting phase.

**Comment by mingmcb** [comment]
yes, it is possible. You can test it out by creating a helm charts with `CleanUpAfterDeletion enabled.  A sample argocd app as shown below can be used for testing, the app will be in stuck if you try to delete.
`apiVersion: "argoproj.io/v1alpha1"
kind: "Application"
metadata:
  finalizers:
    - "resources-finalizer.argocd.argoproj.io"
  name: "pulsar-resources-test"
spec:
  project: applications
  source:
    repoURL: 'https://chartrepo'
    targetRevision: x.x.x
    chart: test-chart
  destination:
    namespace: ns-1
    name: cluster-1`

**Comment by tomjo** [comment]
In our case we are working around this issue by using [argocd sync-waves](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/) to ensure deletion in a specific order
